### PR TITLE
Add Scroll bootnode config

### PR DESCRIPTION
**Pull request by Thegaram**
Contents of this change:
- Configure default bootnodes.
- Configure default genesis block.
- Add `--scroll-alpha` flag.
- Change default sync mode to `full`.

Run a node: `l2geth --scroll-alpha`
Attach to the node: `l2geth --scroll-alpha attach`

### CODE REVIEW

**Review by NazariiDenha — DISMISSED** [review]

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -1633,6 +1644,12 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 		}
 		cfg.Genesis = core.DefaultGoerliGenesisBlock()
 		SetDNSDiscoveryDefaults(cfg, params.GoerliGenesisHash)
+	case ctx.GlobalBool(ScrollAlphaFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 534353
+		}
+		cfg.Genesis = core.DefaultScrollAlphaGenesisBlock()
+		// SetDNSDiscoveryDefaults(cfg, params.ScrollAlphaGenesisHash)
```
Comment (colinlyguo) [inline cmd/utils/flags.go:1652]: Why do we close `SetDNSDiscoveryDefaults`?

**Review by georgehao — DISMISSED** [review]

File: eth/ethconfig/config.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ var LightClientGPO = gasprice.Config{
 
 // Defaults contains default settings for use on the Ethereum main net.
 var Defaults = Config{
-	SyncMode: downloader.SnapSync,
+	SyncMode: downloader.FullSync,
```
Comment (colinlyguo) [inline eth/ethconfig/config.go:64]: Would we add a `TODO` here? Once `SnapShot` is available, we can change it back to `SnapSync`.

File: core/genesis.go [hunk 3]
```diff
@@ -467,7 +479,40 @@ func DeveloperGenesisBlock(period uint64, gasLimit uint64, faucet common.Address
...
+			Code:    account.Code,
+			Storage: s,
+		}
+	}
+
+	return ga, nil
+}
+
 func decodePrealloc(data string) GenesisAlloc {
+	if ga, err := decodePreallocScroll(data); err == nil {
```
Comment (colinlyguo) [inline core/genesis.go:513]: Why don't we just modify the implementation here, or remove the code behind this line?

**Review by colinlyguo — DISMISSED** [review]
LGTM. Leaving some questions.

File: cmd/utils/flags.go (hunk 1 above)
Comment (Thegaram) [reply cmd/utils/flags.go:1652]: This is for DNS-based discovery using hosts like `goerli.ethdisco.net`. We don't have this.

File: eth/ethconfig/config.go (hunk 2 above)
Comment (Thegaram) [reply eth/ethconfig/config.go:64]: Yes, we should change it back to `SnapSync` later.

File: core/genesis.go (hunk 3 above)
Comment (Thegaram) [reply core/genesis.go:513]: The prealloc data for known networks is hardcoded into geth. I added a new format to support `Code` and `Storage`. However, this new format is not compatible with the old format (different RLP encoding). I kept the old format here for compatibility reasons.

Comment (colinlyguo) [reply core/genesis.go:513]: I see now, so if the new one throws an error, then the old decoder is used?
I am curious about in what cases that would happen.

Comment (Thegaram) [reply core/genesis.go:513]: If someone runs `l2geth --goerli` for instance. I guess we don't need to support the old network configs though, might remove in the future.

File: cmd/utils/flags.go (hunk 1 above)
Comment (colinlyguo) [reply cmd/utils/flags.go:1652]: Now I understand this, after reading EIP-1459 (Node Discovery via DNS).

**Pull request merged**